aeppl/rewriting.py

```diff
@@ -282,44 +222,78 @@ def construct_ir_fgraph(
     `FunctionGraph`.
     """
 
+    # We're going to create a `FunctionGraph` that effectively represents the
+    # joint log-probability of all the random variables assigned values in
+    # `rv_values`.
+    # The `FunctionGraph`'s outputs will be `ValuedVariable`s.  This serves to
+    # associate a random variable with its value and facilitate
+    # rewrites/transformations of both simultaneously.
+    # The `FunctionGraph` will be transformed by rewrites so that
+    # log-probabilities can be assigned to/derived for its outputs.
+
+    # We clone the random variables that we'll use as `FunctionGraph` outputs
+    # so that they're distinct nodes in the graph.  This allows us to replace
+    # all instances of the original random variables with their value
+    # variables, while leaving the output clones untouched.
+    rv_value_clones = {}
+    for rv, val in rv_values.items():
+        rv_node_clone = rv.owner.clone()
+        rv_clone = rv_node_clone.outputs[rv.owner.outputs.index(rv)]
+        rv_value_clones[rv_clone] = val
+
+    # We topologically order the `FunctionGraph` outputs just in case
+    # we want to perform operations on them in order (e.g. compute test values).
+    # TODO: It's not clear that this is actually needed.
+    rv_vars = list(rv_value_clones.keys())
+    # sorted_rv_vars: List[TensorVariable] = sum(
+    #     [
+    #         [o for o in node.outputs if o in rv_vars]
+    #         for node in io_toposort((), rv_vars)
+    #     ],
+    #     [],
+    # )
+    sorted_rv_vars = rv_vars
```

Comment:
Is this still necessary?

Reply:
If anything we should perform operations on variables that are conditioned on but are not `RandomVariables` first. Their value variable should be used to set the expression of the value variable of the `RandomVariable` upstream so we can support the examples in https://github.com/aesara-devs/aeppl/discussions/119

Reply:
```python
import aesara.tensor as at
from aeppl import joint_logprob

x = at.normal()
y = at.log(x)
z = at.normal(y)

y_vv = y.clone()
z_vv = z.clone()
joint_logprob({y: y_vv, z: z_vv})
```

In this example we should also replace `x` with a `ValueVariable` whose value variable is set to `x_vv = at.exp(y_vv)`, for instance.